### ISSUE TITLE
[5.0] Incorrect error check in L2CAP channel open

### ISSUE THREAD

**Issue by tomipu** [Polidea/RxBluetoothKit#183]
When opening L2CAP channel, error callback is always executed. Problem is most likely in the error check that should be `error == nil` instead of `error != nil`.

Code snippet from Peripheral.swift:
```
let observable = delegateWrapper.rx_didOpenL2CAPChannel
    .take(1)
    .flatMap { [weak self] (channel, error) -> Observable<CBL2CAPChannel> in
        guard let strongSelf = self else { throw BluetoothError.destroyed }
        if let channel = channel, error != nil {
            return .just(channel)
        } else {
            throw BluetoothError.openingL2CAPChannelFailed(strongSelf, error)
        }
}
```

**Comment by Kacper20** [comment]
Hi @tomipu, you’re entirely right. Thanks so much!

@scibas could you send a patch to this?

**Comment by Kacper20** [comment]
It’s fixed by #184. 

Thanks again for feedback!

**Comment by tomipu** [comment]
Wow, that was fast!
Thanks guys, works like charm now.
